src/vs/workbench/services/languageRuntime/common/languageRuntimeDataExplorerClient.ts

```diff
@@ -496,6 +508,31 @@ export class DataExplorerClientInstance extends Disposable {
 		}
 	}
 
+	async translateToCode(desiredSyntax: string): Promise<TranslatedCode> {
+		const state = await this.getBackendState();
+
+		const columnFilters: Array<ColumnFilter> = state.column_filters;
+		const rowFilters: Array<RowFilter> = state.row_filters;
+		const sortKeys: Array<ColumnSortKey> = state.sort_keys;
+
+		return this.runBackendTask(
+			() => this._backendClient.translateToCode(columnFilters, rowFilters, sortKeys, desiredSyntax),
+			() => ({ 'translated_code': [''] })
+		);
+	}
+
+	/**
+	 * Guess the code syntax for export.
+	 * @returns A promise that resolves to the available code syntaxes.
+	 */
+	async guessCodeSyntax(): Promise<CodeSyntax> {
```

Comment:
Perhaps this is a better place to check that the returned syntax is among those that the backend claims to support.